### ISSUE TITLE
Excessive Strings `<span>` Appears in Viewer 

### ISSUE THREAD

**Issue by chengzhuzhang** [E3SM-Project/e3sm_diags#342]
Could be something in output_viewer that is incompatible with **python 3.8**. (Earlier python versions works fine)
![Screen Shot 2020-09-15 at 12 19 23 AM](https://user-images.githubusercontent.com/13056557/93178463-2530f600-f6e9-11ea-8920-18f0d6d35d16.png)

**Comment by forsyth2** [comment]
@chengzhuzhang Looks like all it took was a one line change -- see https://github.com/ESGF/output_viewer/pull/15. If the attached picture looks good, I can merge that change in `output_viewer`. I suppose we should also release a new version of `output_viewer` then too.

For reference, I did the following steps from `E3SM-Project/e3sm_diags/tests/system`:
```
# Create an environment with python 3.8
conda create -y -n e3sm_diags_env_dev_py38 -c conda-forge -c defaults -c cdat/label/v8.2.1 python=3.8 "cdp>=1.6.0" "cdms2>=3.1.4" "cdutil>=8.2" "genutil>=8.2" "cdtime>=3.1.2" numpy matplotlib cartopy beautifulsoup4 lxml scipy "dask==2.15.0"
# Make changes in ESGF/output_viewer
pip install ../../../../ESGF/output_viewer/ # Install those changes
pip install ../.. # Install E3SM Diags
python -m unittest # Run tests
# Look at the viewer.
```

![viewer_no_span](https://user-images.githubusercontent.com/30700190/99603090-76928580-29b7-11eb-8f4b-463a570c385f.png)

**Comment by chengzhuzhang** [comment]
Looks good! yea we need a new release of output-viewer. I'm curious what change from py3.7to py3.8 caused this...

**Comment by chengzhuzhang** [comment]
I got many lines of errors saying “Bad formatting 'html' is an invalid keyword argument for XMLParser()”
The offending line seems to be  https://github.com/ESGF/output_viewer/blob/f02544ce107e0a3fb9b86afa464a2d363060f84d/output_viewer/htmlbuilder.py#L98
May still need to work on bringing output_viewer to support py3.8, but it is also possible it is an e3sm_diags problem.

**Comment by chengzhuzhang** [comment]
@forsyth2 Would you please look into this? As @xylar said, we'd like to use python 3.8 for the official release of `e3sm-unfied`. So this issue needs to be prioritized. Thanks!

**Comment by forsyth2** [comment]
@chengzhuzhang Sure, but I wasn't able to reproduce the problem locally. I'm assuming this problem doesn't show up in the test unified environment either since the Python version is pinned to be less than 3.8 (https://github.com/conda-forge/output_viewer-feedstock/pull/8/)? In any case, I'll try looking into it more.

**Comment by chengzhuzhang** [comment]
Yes, I also realized it. Would you maybe try a new installation e3sm-diags with py3.8 
```
conda create -y -n e3sm_diags_env_dev_dec1_2020_py3.8 -c conda-forge -c defaults python=3.8   cdp  cdms2  cdutil genutil cdtime numpy matplotlib cartopy beautifulsoup4 lxml scipy  dask
```
On either acme1 or cori, and to see if you can reproduce the problem? My guess is on your local environment, you did additional output_viewer install with pip, which could take some effect.

**Comment by xylar** [comment]
@forsyth2, sorry about https://github.com/conda-forge/output_viewer-feedstock/pull/8.  @chengzhuzhang and i had a conversation and I misunderstood which version needed this constraint.  You should still be able to install `output_viewer` 1.3.2 with py>=3.8 and it will just come from build 0.

**Comment by forsyth2** [comment]
@chengzhuzhang I ran the tests with the `e3sm_diags_env_dev_osx` environment locally, `e3sm_diags_env_dev_dec2_2020_py3.8` environment locally and on Acme1, and `test_e3sm_unified_1.4.0` environment on Cori and Compy. I'm not seeing any errors like `“Bad formatting 'html' is an invalid keyword argument for XMLParser()”`.

**Comment by chengzhuzhang** [comment]
It is expected that `e3sm_diags_env_dev_osx` and `est_e3sm_unified_1.4.0` are fine. But both `e3sm_diags_env_dev_*_py3.8` envs I tested locally and acme1 gave the the errors. Can you try run:
`python all_sets.py -d all_sets.cfg`
under `/tests/system/`

**Comment by forsyth2** [comment]
@chengzhuzhang Ah, that reproduces the problem. I will investigate. Thanks.

**Comment by forsyth2** [comment]
@chengzhuzhang The problem is because we can't specify `html=True` in https://github.com/ESGF/output_viewer/blob/master/output_viewer/htmlbuilder.py#L94 anymore. 

https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.XMLParser states "Changed in version 3.8: Parameters are now keyword-only. The html argument no longer supported."

However, if I just remove that argument, then the leftmost columns in the viewers become blank. Looking into how to get around this issue -- maybe a different parser?

**Comment by forsyth2** [comment]
@chengzhuzhang The viewer appears to render successfully despite that warning -- perhaps it might be best to simply comment it out? 

I also tried `parser = XMLParser(target=proxy, encoding='html')` but that also leaves empty left columns in the viewers....

**Comment by xylar** [comment]
Googling around a bit, it seems like you just want an `HTMLParser` object instead of an `XMLParser`, could that be? https://docs.python.org/3/library/html.parser.html

**Comment by forsyth2** [comment]
I was looking at that, but I don't see a `target` argument to pass in `target=TreeProxy(root)`. Maybe there is a way to include that somehow.

**Comment by xylar** [comment]
Right, I was noticing the same.  Hmm.

**Comment by chengzhuzhang** [comment]
> @chengzhuzhang The viewer appears to render successfully despite that warning -- perhaps it might be best to simply comment it out?
> 
> I also tried `parser = XMLParser(target=proxy, encoding='html')` but that also leaves empty left columns in the viewers....

In this case, i think it is okay to just comment out the printing statement.

**Comment by chengzhuzhang** [comment]
`e3sm_diags` calls `cdp_viewer` which then calls `output_viewer`.  https://github.com/CDAT/cdp/blob/master/jupyter/using-the-cdp-viewer.ipynb this is what I found for cdp_viwer. And `output_viewer` is not really documented. I'm a little concerned that lack of good understanding of `output_viewer` will give more troubles in the future.

**Comment by forsyth2** [comment]
@chengzhuzhang Thanks, I will comment out the print statement then. And yes I'm also concerned about `output_viewer`. Perhaps when I have time I could study how it works and add code comments. That said, this particular issue is more rooted in the fact that XMLParser in Python 3.8 is not backwards compatible....

**Comment by forsyth2** [comment]
@chengzhuzhang @xylar I released v1.3.3 of `output_viewer`: https://github.com/ESGF/output_viewer/releases. It just comments out that warning, since the `XMLParser` exception handling appears to still render the viewer correctly.
